### PR TITLE
fix: manifest validator should add a warning instead of an error on non-square svg icons

### LINKED ISSUE #3322: "Icons must be square." for svg icons
### Describe the problem and steps to reproduce it:

I'm trying to upload a new version of [cf-pop](https://addons.mozilla.org/firefox/addon/cf-pop/) to amo.

### What happened?

It says `Icons must be square.` and `Icon at "icons/Cloudflare-gray.svg" must be square.` twice.

### What did you expect to happen?

It should accept svg icons (like before).

### Anything else we should know?

The icon file didn't change. It's exactly the same icon file in the published version.

### PR DESCRIPTION
**Pull request by Yrkan**
Fixes #3322

it's now a warning instead of an error.

### CODE REVIEW

**Comment by wagnerand** [comment]
@ThamiMemel Can you fix the failing tests?

**Comment by ThamiMemel** [comment]
Yes, sure i'll change it to adds a warning if the dimensions of the image are not the same instead of an error, please correct me if i'm wrong here.

**Comment by rpl** [comment]
@wagnerand @ThemiMemel the way I read the issue described in #3322 is that the addons-linter should only warn that the icon is not square if the image is an svg, but still mark it as an error if it is not an svg (e.g. if it is a png image).

Basically this seems to be an addition to the fix applied for #1669 (which was about not warning about the size not matching the convention that Firefox expects for the icon image size), which was fixed in #1689 by adding a `info.mime !== 'image/svg+xml'` check to the validation that follows this one.

And so in my opinion we should check if the image is an svg or not (using a check similar to that `info.mime !== 'image/svg+xml'`) and:
- add it as a warning if the image is an svg
- still add it as error if the image is not an svg

Then keep the existing test unmodified and add a new test case that verifies that if the size mismatch is triggered by an svg icon image the validation failure is collected as a warning and that as a result the manifest is considered valid and the linting passes.

**Comment by ThamiMemel** [comment]
@rpl Oh my bad, looks like the changes to test and update the validation to work as intended.
Thank you!  i'll send a PR with the changes.

**Comment by rpl** [comment]
> @rpl Oh my bad, looks like the changes to test and update the validation to work as intended.

No worries at all, on the contrary my sincere apologies for missing to notice this PR before, I did notice it today when Andreas did comment on it.

> Thank you! i'll send a PR with the changes.

Update this PR without creating a new pull request.

If you have never done that, there are multiple way to achieve that, e.g. you may choose one of the following strategies (the first is likely the easiest one if you have never used `git rebase -i ...` or `git push ... --force`):

- add more commit to the same branch, where you revert the parts that didn't need to change and apply further changes to the one that needs to be adapted, and then just push the same branch in your github fork
- do an interactive git rebase on the branch  and drop the commit that you don't need, then add new commit with further changes, and then push with --force the same branch in your github fork (--force it is needed because there are not only additional commits and the branch has to be changed to point to the new set of commits in your local branch)
- remove the local branch, and then create a new branch with the same name, make the change that you need and then push with --force in your github fork (--force is is needed for the same reason described above)

**Comment by ThamiMemel** [comment]
@rpl  yeah i was autopiloting when i said i'll send a PR with the changes i meant i'll commit to this PR.

P.S: the commit is already here, please let me know if anything need to be changed.
Thank you again.

File: src/parsers/manifestjson.js [hunk 1]
```diff
@@ -510,8 +510,12 @@ export default class ManifestJSONParser extends JSONParser {
...
       const info = await getImageMetadata(this.io, iconPath);
       if (info.width !== info.height) {
-        this.collector.addError(messages.iconIsNotSquare(iconPath));
-        this.isValid = false;
+        if (info.mime !== 'image/svg+xml') {
+          this.collector.addError(messages.iconIsNotSquare(iconPath));
+          this.isValid = false;
+        } else {
+          this.collector.addWarning(messages.iconIsNotSquare(iconPath));
+        }
```
Comment (rpl) [inline src/parsers/manifestjson.js:518]: :+1: this change on the implementation side looks the one I was thinking of.

This PR should only be missing a new test case now, would you mind to define a new test case to cover the warning on non square svg icons?

you could start by looking to the test case that you were changing in the previous version of this PR, and based on that create a new test case to cover the new expected behavior.

**Review by rpl — CHANGES_REQUESTED** [review]

File: src/parsers/manifestjson.js (hunk 1 above)
Comment (ThamiMemel) [reply src/parsers/manifestjson.js:518]: Sure, i have already made one, i'll commit it right now, please correct me if anything is wrong. Thank you.

File: tests/unit/parsers/test.manifestjson.js [hunk 2]
```diff
@@ -1730,6 +1730,31 @@ describe('ManifestJSONParser', () => {
         expect(warnings.length).toEqual(0);
       });
 
+      it('adds a warning if the icon is SVG but the dimensions of the image are not the same', async () => {
```
Comment (rpl) [inline tests/unit/parsers/test.manifestjson.js:1733]: Nitpick: let's make this test description a bit shorter e.g. `it('does add a warning if the icon is a non square svg', ...)`

**Review by rpl — APPROVED** [review]
@ThamiMemel thanks! 

This looks good to me and I'm marking this PR as accepted, it would be nice to shorten the description of the new test as described below as a small nitpick to cover before landing this.

File: tests/unit/parsers/test.manifestjson.js (hunk 2 above)
Comment (ThamiMemel) [reply tests/unit/parsers/test.manifestjson.js]: done.
Thank for the feedback

**Review by rpl — APPROVED** [review]

**Comment by caitmuenster** [comment]
Thanks so much for the patch, @Yrkan! 🙌 Your contribution has been added to our [recognition wiki](https://wiki.mozilla.org/Add-ons/Contribute/Recognition). 

Welcome onboard -- we look forward to seeing you around!

**Pull request merged**